File: keeperFinal/Models/Keep.cs
namespace keeperFinal.Models;

public class Keep : IHasCreator
{ 
  public int Id { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int? views { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public int? Kept { get; set; }
}

File: keeperFinal.client/src/models/Keep.js
import { Profile } from "./Profile";


export class Keep {
  constructor(data) {
    this.id = data.id;
    this.views = data.views;
    this.kept = data.kept;
    this.img = data.img;
    this.name = data.name;
    this.description = data.description;
    this.creatorId = data.creatorId;
    this.creator = new Profile(data.creator);
    this.createdAt = new Date(data.createdAt).toLocaleDateString();
    this.updatedAt = new Date(data.updatedAt).toLocaleDateString();
    this.isKept = false
  }
}
